### PR TITLE
VC-4247: added Withdraw state

### PR DESCRIPTION
**Pull request by at-kh**
## Description

Closes: #XXXX

---

### Author Checklist

*All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues.*

I have...

- [ ] included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] added `!` to the type prefix if API or client breaking change
- [ ] targeted the correct branch
- [ ] provided a link to the relevant issue or specification
- [ ] added a changelog entry to `CHANGELOG.md`
- [ ] included comments for [documenting Go code](https://blog.golang.org/godoc)
- [ ] updated the relevant documentation or specification
- [ ] reviewed "Files changed" and left comments if necessary
- [ ] confirmed all CI checks have passed

### Reviewers Checklist

*All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items.*

I have...

- [ ] confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] confirmed `!` in the type prefix if API or client breaking change
- [ ] confirmed all author checklist items have been addressed
- [ ] reviewed API design and naming
- [ ] reviewed documentation is accurate
- [ ] reviewed tests and test coverage
- [ ] manually tested (if applicable)

### CODE REVIEW

File: database/vipcoin/chain/banking/banking.go [hunk 1]
```diff
@@ -67,12 +67,8 @@ func (r Repository) UpdateBaseTransfers(transfers ...*bankingtypes.BaseTransfer)
 				tx_hash = :tx_hash
 			 WHERE id = :id`
 
-	for _, transfer := range transfers {
-		transferDB := toTransferDatabase(transfer)
-
-		if _, err := tx.NamedExec(query, transferDB); err != nil {
-			return err
-		}
+	if _, err := tx.NamedExec(query, toTransfersDatabase(transfers...)); err != nil {
```
Comment (alex-bodnar) [inline database/vipcoin/chain/banking/banking.go]: С update так не сработает, он ошибку выдаст если в масиве transfers будет больше одного элемента

File: database/vipcoin/chain/banking/issue.go [hunk 2]
```diff
@@ -90,16 +90,14 @@ func (r Repository) UpdateIssues(issues ...*bankingtypes.Issue) error {
 	WHERE id =:id;
 	`
 
-	for _, issue := range issues {
-		issueDB := toIssuesDatabase(issue)
+	issueDB := toIssuesDatabase(issues...)
```
Comment (alex-bodnar) [inline database/vipcoin/chain/banking/issue.go]: С update так не сработает, он ошибку выдаст если в масиве issues будет больше одного элемента

File: database/vipcoin/chain/banking/payment.go [hunk 3]
```diff
@@ -93,16 +90,14 @@ func (r Repository) UpdatePayments(payments ...*bankingtypes.Payment) error {
 	WHERE id =:id;
 	`
 
-	for _, payment := range payments {
-		paymentDB := toPaymentDatabase(payment)
+	paymentDB := toPaymentsDatabase(payments...)
```
Comment (alex-bodnar) [inline database/vipcoin/chain/banking/payment.go]: С update так не сработает, он ошибку выдаст если в масиве payments будет больше одного элемента

File: database/vipcoin/chain/banking/system_transfer.go [hunk 4]
```diff
@@ -90,16 +90,14 @@ func (r Repository) UpdateSystemTransfers(transfers ...*bankingtypes.SystemTrans
 	WHERE id =:id;
 	`
 
-	for _, transfer := range transfers {
-		transferDB := toSystemTransfersDatabase(transfer)
+	transferDB := toSystemTransfersDatabase(transfers...)
```
Comment (alex-bodnar) [inline database/vipcoin/chain/banking/system_transfer.go]: С update так не сработает, он ошибку выдаст если в масиве transfers будет больше одного элемента

File: database/vipcoin/chain/banking/withdraw.go [hunk 5]
```diff
@@ -1,21 +1,104 @@
...
+	id =:id, asset =:asset, amount =:amount, kind =:kind, extras =:extras, timestamp =:timestamp, tx_hash =:tx_hash
+	WHERE id =:id;
+	`
+
+	queryWithdraw := `UPDATE vipcoin_chain_banking_withdraw SET
+	id =:id, wallet =:wallet
+	WHERE id =:id;
+	`
+
+	withdrawsDB := toWithdrawsDatabase(withdraws...)
```
Comment (alex-bodnar) [inline database/vipcoin/chain/banking/withdraw.go]: С update так не сработает, он ошибку выдаст если в масиве withdraws будет больше одного элемента

**Review by alex-bodnar — APPROVED** [review]
LGTM

**Comment by vasylNaumenko** [comment]
**LGTM**

**Pull request merged**